docs/README.md

```diff
@@ -0,0 +1,56 @@
+## Sequence Diagram
+
+This is how the user interacts with the various components of this application.
+
+The different components have been chosen with specific needs in mind:
+* Backend: The backend is needed to protect the private key of the wallet that is doing the actual signing of the transactions on the blockchain.
+* Forwarder: Ensures that the incoming transaction gets validated before actually sent for execution. The forwarder also keeps a record of the internal nonces that external wallets are using while forwarding.
+* Recipient: This is the final smart contract whose method is originally called from the Frontend. It substitutes the forwarder sender for the actual sender (from) of the original transaction and persists the data on-chain.
+
+```mermaid
+  sequenceDiagram
+    participant Frontend
+    participant Metamask
+    participant User
+    participant Backend
+    participant Forwarder
+    participant Recipient
+
+    rect rgb(111, 160, 200, 0.8)
+        Frontend-->>Recipient: Subscribe to Interaction Events
+    end
+
+    activate Metamask
+    rect rgb(208, 240, 192)
+      Frontend->>Metamask: Request Accounts
+      Metamask->>User: Ask to select wallet <br /> and connect
+      User->>Frontend: selects wallet and connects
+    end
+    rect rgb(208, 240, 192)
+      Frontend->>Metamask: Request to INTERACT
+      Metamask->>User: Popup and request to sign a message <br /> (the message contains the final INTERACT step <br/> and the current nonce for the User)
+      User->>Frontend: Signs message
+    end
+    deactivate Metamask
+    activate Backend
+
+    rect rgb(208, 160, 200, 0.9)
+      Frontend->>Backend: Send signed message and sender
+      Backend->>Backend: Validate that the User (original signer)<br /> has actually signed the message
```

Comment:
This verification is a mroe broad and generalized verification. It is not as precise as the one made on the chain by the verify function. I tested some cases where this would be okay but the one on the chain would throw an error. Thats why im debating with me to leave this in or not. But it might make sense to make a first small verification to aboart the process if necessary early.

Reply:
This is a very vague explanation here; you are basically saying, that you don't know exactly what the difference between the two validation methods is and why one can succeed where the other fails. This is a clear sign that we need to invest more time to understand how the 2 validation methods function under the hood :)

My own take on this was rather that the BE needs to validate before signing to ensure that the frontend has not been tampered with and then the SC on the next step validates again to ensure that the backend has not bee n tampered with. So we probably need both validations. But if you are not sure, maybe we need to take another look at the relay documentation. Let me know.